custom_components/fellow/fellow_aiden/__init__.py

```
@@ -325,6 +349,95 @@ def get_brewer_id(self) -> str | None:
+
+    async def _probe_list_resource(
+        self, brewer_id: str, url: str, action: str
+    ) -> list[dict[str, Any]]:
+        """Probe a candidate device endpoint and treat non-Aiden shapes as incompatible."""
+        response = await self._request_with_reauth("get", url)
+
+        if response.status in (401, 403):
+            parsed = await self._parse_response(response)
+            raise FellowAuthError(
+                f"{action} unauthorized for device {brewer_id} ({response.status}): {parsed}"
+            )
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)